Card is invalid if nil or empty.

gavinlaking opened the pull request:
Previously, nil, or an empty string could be given to `CardValidation.valid?` and would be accepted as a valid card number. This pull request corrects this behaviour.

gavinlaking committed:
Add .ruby-version file, set to ruby-2.2.

gavinlaking committed:
Card is invalid if nil or empty.

gavinlaking committed:
Bump gem version to 1.1.0.

craigbrad commented on lib/card_validation/validity.rb:
```diff
@@ -51,6 +51,14 @@ def double
   end
 
+  def empty?
+    number.respond_to?(:empty?) && number.empty?
```
Instead of checking if number responds to `empty?`, would it be beneficial to make sure number returns an object you can call `empty?` on?

gavinlaking replied:
Good catch! https://github.com/onthebeach/card_validation/pull/2/commits/6738648b60af6fbc7426d58d5285ef73933ebf49 fixes this.

gavinlaking committed:
Code review feedback;

Previously the number was being checked to see if it responded to the 'empty?'
message. Instead, we can convert the number to a string then send the 'empty?'
message; simplifying this validation.

craigbrad commented:
LGTM 👍

jondoveston commented:
👍 LGTM
